dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/api/imports/ImportDDIServiceBeanTest.java

```diff
@@ -1,20 +1,628 @@
+        // given
+        String ddiXml = "<codeBook>"
+                + "<otherMat ID=\"f15\" URI=\"https://doi.org/10.1012/someId\" level=\"datafile\">"
+                + "<labl>file.txt</labl>"
+                + "<txt>some description</txt>"
+                + "<notes level=\"file\" type=\"DATAVERSE:CONTENTTYPE\" subject=\"Content/MIME Type\">text/plain</notes>"
+                + "</otherMat>"
+
+                + "<otherMat ID=\"f16\" URI=\"https://doi.org/10.1012/someOtherId\" level=\"datafile\">"
+                + "<labl>file2.pdf</labl>"
+                + "<txt>some description2</txt>"
+                + "<notes level=\"file\" type=\"DATAVERSE:CONTENTTYPE\" subject=\"Content/MIME Type\">application/pdf</notes>"
+                + "</otherMat>"
+                + "</codeBook>";
+
+        // when
+        DatasetDTO datasetDto = importDdiService.doImport(ImportType.HARVEST, ddiXml);
+
+        // then
+        DatasetVersionDTO datasetVersionDto = datasetDto.getDatasetVersion();
+
+        assertThat(datasetVersionDto.getFiles()).hasSize(2)
+                                                .extracting(
+                                                        FileDTO::getLabel,
+                                                        FileDTO::getDescription,
+                                                        f -> f.getDataFile().getContentType(),
+                                                        f -> f.getDataFile().getStorageIdentifier())
+                                                .contains(
+                                                        tuple("file.txt", "some description", "text/plain", "https://doi.org/10.1012/someId"),
+                                                        tuple("file2.pdf", "some description2", "application/pdf", "https://doi.org/10.1012/someOtherId"));
+    }
+
+    @Test
+    void doImport_should_parse_files_info_from_fileDscr() throws XMLStreamException, ImportException {
+        // given
+        String ddiXml = "<codeBook>"
+                + "<fileDscr ID=\"f15\" URI=\"http://localhost:8080/api/access/datafile/15\">"
+                + "<fileTxt>"
+                + "<fileName>file.txt</fileName>"
+                + "<dimensns><caseQnty>100</caseQnty><varQnty>3</varQnty><recPrCas>10</recPrCas></dimensns>"
+                + "<fileType>text/plain</fileType>"
+                + "</fileTxt></fileDscr>"
+
+                + "<fileDscr ID=\"f16\" URI=\"http://localhost:8080/api/access/datafile/16\">"
+                + "<fileTxt>"
+                + "<fileName>file2.pdf</fileName>"
+                + "<fileType>application/pdf</fileType>"
+                + "</fileTxt></fileDscr>"
+                + "</codeBook>";
+        // when
+        DatasetDTO datasetDto = importDdiService.doImport(ImportType.HARVEST, ddiXml);
+
+        // then
+        DatasetVersionDTO datasetVersionDto = datasetDto.getDatasetVersion();
+
+        assertThat(datasetVersionDto.getFiles()).hasSize(2)
```

Comment:
Instead of size checking, we could use `containsExactly` or `containsExactlyInAnyOrder`, but it's ok as is.